docs/reference/search/request/highlighting.asciidoc

```diff
@@ -45,35 +43,43 @@ from versions before 5.0) that match the expression to be highlighted.
 
-The default choice of highlighter is of type `plain` and uses the Lucene highlighter.
-It tries hard to reflect the query matching logic in terms of understanding word importance and any word positioning criteria in phrase queries.
+The default choice of highlighter is of type `unified` and uses the Lucene Unified highlighter.
 
-[WARNING]
-If you want to highlight a lot of fields in a lot of documents with complex queries this highlighter will not be fast.
-In its efforts to accurately reflect query logic it creates a tiny in-memory index and re-runs the original query criteria through
-Lucene's query execution planner to get access to low-level match information on the current document.
-This is repeated for every field and every document that needs highlighting. If this presents a performance issue in your system consider using an alternative highlighter.
+This highlighter breaks the text into sentences and scores individual sentences as
+if they were documents in this corpus, using the BM25 algorithm.
+It also supports accurate phrase and multi-term (fuzzy, prefix, regex) highlighting.
+
+[float]
+===== Offsets Strategy
 
-[[postings-highlighter]]
-==== Postings highlighter
+The `unified` highlighter adapts its strategy to extract the matching offsets in the text to highlight
+depending on the index options set on the field and the type of the query.
+In this highlighter, offsets can be extracted from either postings , term vectors, or via re-analyzing text.
 
-If `index_options` is set to `offsets` in the mapping the postings highlighter
-will be used instead of the plain highlighter. The postings highlighter:
+[float]
+====== Plain highlighting
+
+This mode is picked when there is no other alternative.
+It creates a tiny in-memory index and re-runs the original query criteria through
+Lucene's query execution planner to get access to low-level match information on the current document.
+This is repeated for every field and every document that needs highlighting.
 
-* Is faster since it doesn't require to reanalyze the text to be highlighted:
-the larger the documents the better the performance gain should be
-* Requires less disk space than term_vectors, needed for the fast vector
-highlighter
-* Breaks the text into sentences and highlights them. Plays really well with
-natural languages, not as well with fields containing for instance html markup
-* Treats the document as the whole corpus, and scores individual sentences as
-if they were documents in this corpus, using the  BM25 algorithm
+[float]
+====== Postings
+
+If `index_options` is set to `offsets` in the mapping the `unified` highlighter
+will use this information to highlight documents without re-analyzing the text.
+It re-runs the original query directly on the postings and extracts the matching offsets
+directly from the index limiting the collection to the highlighted documents.
+This mode is faster since it doesn't require to reanalyze the text to be highlighted
```

Comment:
I expect it isn't actually faster for very short strings. At least, that was my experience with the experimental highlighter.